### PR TITLE
Maven for java port

### PR DESCRIPTION
**Pull request by kag0**
Set up maven directory structure and pom.xml
Made minify(String) method static
Made Minify class abstract
Split test cases into their own junit tests
Cleaned up other misc items.

### CODE REVIEW

File: src/main/java/com/getify/minify/Minify.java [hunk 1]
```diff
@@ -0,0 +1,103 @@
+//
+//  Minify.java
+//  jsonminifier - Ported from https://github.com/getify/JSON.minify
+//
+//  Created by Bernhard Gass on 8/01/13.
+//  Copyright © 2013 Bernhard Gass. All rights reserved.
+
+package com.getify.minify;
```
Comment (pradyunsg) [inline src/main/java/com/getify/minify/Minify.java]: I think `com.json_minify` is what was decided the last time around. (In #27)

File: src/test/java/com/getify/minify/MinifyTest.java [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+/**
+ * Created by nfischer on 11/28/15.
+ */
+public class MinifyTest {
+
+    @Test
+    public void testBasicComments(){
+        System.out.println("Running test1...");
+        String test1 = "// this is a JSON file with comments\n" +
```
Comment (pradyunsg) [inline src/test/java/com/getify/minify/MinifyTest.java]: Possibly change this and following lines from 

```
String test1 = "..." +
``` 

to 

```
String test1 = "" +
    "..." +

```

**Comment by pradyunsg** [comment]
If we are adding maven, can we also be running a linter (checkstyle?) on this. Something to stick with say a Style Guide (I'd say Google Java Style Guide).

File: src/main/java/com/getify/minify/Minify.java (hunk 1 above)
Comment (kag0) [inline src/main/java/com/getify/minify/Minify.java]: @pradyunsg that can be changed, but if we want to use `com.json_minify` then we really should(/must) obtain the domain `json_minify.com`. The reason for this is that java package naming convention is to use the reversed internet domain as the package name. The low level domain `minify` for this project would correspond to `minify.getify.com`, while that domain might not actually exist, it's ok, because we control the second level domain `getify.org`.

**TL;DR:** Java convention for package naming is to use the reversed domain of the organization, followed by the project name.

**Comment by kag0** [comment]
@pradyunsg I recently found [codacity](https://www.codacy.com/) which could be interesting. Or we could run some linter on CI. Is that request specific to this PR or maven in particular or can it be opened as a standalone issue?

File: src/main/java/com/getify/minify/Minify.java (hunk 1 above)
Comment (getify) [inline src/main/java/com/getify/minify/Minify.java]: Seems silly to me. What about `http://getify.github.io/` -> `io.github.getify`?

Comment (kag0) [inline src/main/java/com/getify/minify/Minify.java]: @getify: yup, `io.github.getify`would be a valid package prefix and maven `groupId`. It's fairly common, especially among organizations/groups who are committed to github or are on a razor budget that doesn't include domain names.

Comment (getify) [inline src/main/java/com/getify/minify/Minify.java]: `getify.com` is softly reserved in case my company ever had some sort of corporate activity or interest. The OSS work I do (on github) is merely a side project in that respect.

Since this project is not likely to ever have its own site/domain name, and will thus always live on github, I think it's best to style the package namespacing in that way.

Comment (kag0) [inline src/main/java/com/getify/minify/Minify.java]: Ah, I see. In that case, do you see yourself releasing more java code under `io.github.getify` and if so, is it OK for those projects to be associated under the same maven group as this project?

Comment (getify) [inline src/main/java/com/getify/minify/Minify.java]: I don't really do any Java work. The only reason this is a Java project at all is as a language port of the original implementation. I think it's very unlikely that I'll do future Java work, but it's not entirely implausible that some other project might also have a Java port in a similar vein as this one. Even still, I am only the host of that code, it's not code I write or actively maintain.

Comment (kag0) [inline src/main/java/com/getify/minify/Minify.java]: `io.github.getify` should work fine then. I'll get that changed and we can get this PR merged.

**Comment by kag0** [comment]
@getify @pradyunsg 
I believe this should be ready to merge. Are there any blocking issues?

**Comment by pradyunsg** [comment]
Not to me.

**Comment by getify** [comment]
Not that I would know the difference, but LGTM.
